clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.deploy;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.clouddriver.security.resources.AccountNameable;
+import com.netflix.spinnaker.clouddriver.security.resources.ApplicationNameable;
+import com.netflix.spinnaker.clouddriver.security.resources.ResourcesNameable;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.Errors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class DescriptionAuthorizer<T> {
+  private final Registry registry;
+  private final FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  private final Id missingApplicationId;
+
+  public DescriptionAuthorizer(Registry registry,
+                               Optional<FiatPermissionEvaluator> fiatPermissionEvaluator) {
+    this.registry = registry;
+    this.fiatPermissionEvaluator = fiatPermissionEvaluator.orElse(null);
+
+    this.missingApplicationId = registry.createId("authorization.missingApplication");
+  }
+
+  public void authorize(T description, Errors errors) {
+    if (fiatPermissionEvaluator == null) {
+      return;
+    }
+
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+
+    String account = null;
+    List<String> applications = new ArrayList<>();
+
+    if (description instanceof AccountNameable) {
+      AccountNameable accountNameable = (AccountNameable) description;
+      account = accountNameable.getAccount();
+    }
+
+    if (description instanceof ApplicationNameable) {
+      ApplicationNameable applicationNameable = (ApplicationNameable) description;
+      applications.addAll(applicationNameable.getApplications());
+    }
+
+    if (description instanceof ResourcesNameable) {
+      ResourcesNameable resourcesNameable = (ResourcesNameable) description;
+      applications.addAll(resourcesNameable.getResourceApplications());
+    }
+
+
+    if (account != null && !fiatPermissionEvaluator.hasPermission(auth, account, "ACCOUNT", "WRITE")) {
+      errors.reject("authorization", format("Access denied to account %s", account));
+    }
+
+    if (!applications.isEmpty()) {
+      fiatPermissionEvaluator.storeWholePermission();
+
+      applications.forEach(application -> {
+        if (!fiatPermissionEvaluator.hasPermission(auth, application, "APPLICATION", "WRITE")) {
+          errors.reject("authorization", format("Access denied to application %s", application));
+        }
+      });
+    }
+
+
+    if (account != null && applications.isEmpty()) {
+      registry
```

Comment:
Will also log a suitable message ... as mentioned in the PR description I'm looking to get an idea of how many descriptions are account restricted but not application-restricted.

Likely also worth a `success` tag on this counter.

The plan would be to mark the descriptions that legitimately have no applications, and fix the others.